buildstock_fetch/utils/ev_utils.py

```diff
@@ -0,0 +1,297 @@
+    metadata_df = metadata_df.with_columns([
+        pl.when(pl.col("occupants") == "10+").then(pl.lit("10")).otherwise(pl.col("occupants")).alias("occupants_temp")
+    ])
+
+    # Now cast to numeric
+    metadata_df = metadata_df.with_columns([pl.col("occupants_temp").cast(pl.Int64).alias("occupants")]).drop(
+        "occupants_temp"
+    )
+
+    # # Create household size bins
+    # metadata_df = metadata_df.with_columns([
+    #     pl.when(pl.col("occupants") > 1)
+    #     .then(pl.lit("2+ ppl"))
+    #     .when(pl.col("occupants") == 1)
+    #     .then(pl.lit("1 person"))
+    #     .otherwise(pl.lit(None))
+    #     .alias("hhsize_bins")
+    # ])
+
+    # Process income categories - convert to standard ranges first
+    metadata_df = metadata_df.with_columns([
+        pl.when(pl.col("income") == "<10000")
+        .then(pl.lit("0-10000"))
+        .when(pl.col("income") == "200000+")
+        .then(pl.lit("200000-400000"))
+        .when(pl.col("income") == "Not Available")
+        .then(pl.lit(None))
+        .otherwise(pl.col("income"))
+        .alias("income")
+    ])
+
+    # Convert income ranges to numeric midpoints
+    metadata_df = metadata_df.with_columns([pl.col("income").map_elements(assign_income_midpoints).alias("income")])
+
+    # Extract last 5 characters from PUMA
+    metadata_df = metadata_df.with_columns([pl.col("puma").str.slice(-5).alias("puma")])
+
+    return metadata_df
+
+
+def load_nhts_data(nhts_path: str, state: str) -> pl.DataFrame:
+    """
+    Load and preprocess the NHTS trip data for a specific state.
+
+    Args:
+        nhts_path: Path to the NHTS trip data file
+        state: State abbreviation to filter data for
+
+    Returns:
+        DataFrame with trip records filtered for the specified state's census division
+    """
+    if not Path(nhts_path).exists():
+        msg = f"NHTS file not found: {nhts_path}. Please run `just download-nhts` to download the data."
+        raise FileNotFoundError(msg)
+
+    # Define the columns we need
+    needed_columns = [
+        "CENSUS_D",  # census division (needed for filtering)
+        "VEHCASEID",  # unique hh/vehicle id
+        "STRTTIME",  # start time
+        "ENDTIME",  # end time
+        "TRPMILES",  # miles driven
+        "TDWKND",  # weekday(2)/weekend(1) flag
+        "HHSIZE",  # occupants
+        "HHFAMINC",  # household income
+        "HHVEHCNT",  # total number of vehicles
+        "URBRUR",  # urban/rural status urban(1)/rural(2)
+    ]
+
+    # Load only the needed columns
+    nhts_df = pl.read_csv(nhts_path, columns=needed_columns)
+
+    # Get the census division for this state
+    state_division = get_census_division_for_state(state)
+
+    # Filter to only keep census division for this state
+    nhts_df = nhts_df.filter(pl.col("CENSUS_D") == state_division)
+
+    # Remove the CENSUS_D column since we don't need it anymore
+    nhts_df = nhts_df.drop("CENSUS_D")
+
+    nhts_df = nhts_df.rename({
```

Comment:
Maybe define this dictionary when `needed_columns` is now, keeping your nice comments, and extract `needed_columns` with it?